tests/components/MapboxMap.test.tsx

```diff
@@ -0,0 +1,112 @@
+      },
+      source: Source.KnowledgeManager
+    }]
+  }
+};
+
+describe('default "getCoordinate"', () => {
+  it('uses result\'s "yextDisplayCoordinate" for marker location', () => {
+    mockAnswersState(mockedStateDefaultCoordinate);
+    const setLngLat = jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
+    render(<MapboxMap mapboxAccessToken='TEST_KEY' />);
+    expect(setLngLat).toBeCalledWith({ lat: 1, lng: 1 });
+  });
+
+  it('displays an error when "yextDisplayCoordinate" field is not present in result', () => {
+    mockAnswersState(mockedStateCustomCoordinate);
+    const setLngLat = jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
+    const errorSpy = jest.spyOn(console, 'error').mockImplementation();
+    render(<MapboxMap mapboxAccessToken='TEST_KEY' />);
+    expect(errorSpy).toBeCalledTimes(1);
+    const expectedMessage = 'Unable to use the default "yextDisplayCoordinate" field as the result\'s coordinate';
+    expect(errorSpy).toBeCalledWith(expect.stringContaining(expectedMessage));
+    expect(setLngLat).not.toBeCalled();
+  });
+
+  it('displays an error when "yextDisplayCoordinate" field is not of type "Coordinate"', () => {
+    mockAnswersState(mockedStateWrongCoordinateType);
+    const setLngLat = jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
+    const errorSpy = jest.spyOn(console, 'error').mockImplementation();
+    render(<MapboxMap mapboxAccessToken='TEST_KEY' />);
+    expect(errorSpy).toBeCalledTimes(1);
+    const expectedMessage = 'The default `yextDisplayCoordinate` field from result is not of type "Coordinate".';
+    expect(errorSpy).toBeCalledWith(expect.stringContaining(expectedMessage));
+    expect(setLngLat).not.toBeCalled();
+  });
+});
+
+it('executes custom "getCoordinate" and use the derived coordinate for marker location', () => {
+  mockAnswersState(mockedStateCustomCoordinate);
+  const setLngLat = jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
+  const customGetCoordinate: CoordinateGetter<Location> = jest.fn().mockImplementation(
+    result => result.rawData.customCoordinate);
+
+  render(<MapboxMap mapboxAccessToken='TEST_KEY' getCoordinate={customGetCoordinate} />);
+  expect(customGetCoordinate).toBeCalledTimes(1);
+  expect(setLngLat).toBeCalledWith({ lat: 2, lng: 2 });
+});
+
+it('registers "onDrag" callback to Mapbox\'s event listener for "drag to pan" interaction', () => {
+  jest.useFakeTimers();
+  jest.spyOn(Marker.prototype, 'setLngLat').mockReturnValue(Marker.prototype);
+  const mapOnEventListener = jest.spyOn(Map.prototype, 'on')
+    .mockImplementation((e, cb) => {
+      e === 'drag' && cb({});
+      return Map.prototype;
+    });
+  const onDragFn = jest.fn().mockImplementation(() => console.log('HERE'));
+
+  render(<MapboxMap mapboxAccessToken='TEST_KEY' onDrag={onDragFn} />);
+  expect(mapOnEventListener).toBeCalledWith('drag', expect.anything());
```

Comment:
does a map drag automatically occur when it's rendered for the first time?

Reply:
no

Reply:
maybe I'm missing something then. what causes the event listener to be called then?

Reply:
since mapbox-gl is mocked, I can't simulate the actual drag event. So, I mocked `Map.on` event listener register to immediately invoked the callback function if the event type is 'drag' instead of waiting for a drag event to occur. Assuming that mapbox-gl's Map.on works as expected, we are testing in our code whether we register the event listener properly when initialize the map and if callback with the debounce time is executed as expected